src/main/java/com/nirima/jenkins/plugins/docker/utils/PortUtils.java

```diff
@@ -65,6 +65,7 @@ public boolean executeOnce() {
             try (Socket ignored = new Socket(host, port)) {
                 return true;
             } catch (IOException e) {
+                LOGGER.warn("Socket to host: " + host + " with port: " + String.valueOf(port) + " couldn't be opened", e);
```

Comment:
This is going to spam the log with nasty exceptions even in normal operating conditions.
While I can understand why you might want to have added this during dev work, it can't go into the core code as-is.

FYI _normally_ the docker plugin will run `executeOnce` multiple times after starting the docker container, waiting for the port to open.  If it doesn't open then we get a _single_ log message about it.
I don't want to fill the log full of exception traces just because the socket wasn't open before we started looking.

TL;DR: Please remove this line,